src/libraries/System.Memory/src/System/Buffers/SequenceReader.Search.cs

```diff
@@ -405,6 +405,42 @@ private bool TryReadToAnyInternal(out ReadOnlySequence<T> sequence, ReadOnlySpan
             return false;
         }
 
+        /// <summary>
+        /// Try to read everything up to the given <paramref name="delimiter"/>.
+        /// </summary>
+        /// <param name="span">The read data, if any.</param>
+        /// <param name="delimiter">The delimiter to look for.</param>
+        /// <param name="advancePastDelimiter">True to move past the <paramref name="delimiter"/> if found.</param>
+        /// <returns>True if the <paramref name="delimiter"/> was found.</returns>
+        public bool TryReadTo(out ReadOnlySpan<T> span, ReadOnlySpan<T> delimiter, bool advancePastDelimiter = true)
+        {
+            ReadOnlySpan<T> remaining = UnreadSpan;
+            int index = remaining.IndexOf(delimiter);
+
+            if (index >= 0)
+            {
+                span = remaining.Slice(0, index);
+                AdvanceCurrentSpan(index + (advancePastDelimiter ? delimiter.Length : 0));
+                return true;
+            }
+
+            // This delimiter might be skipped, go down the slow path
+            return TryReadToSlow(out span, delimiter, advancePastDelimiter);
+        }
+
+        private bool TryReadToSlow(out ReadOnlySpan<T> span, ReadOnlySpan<T> delimiter, bool advancePastDelimiter)
```

Comment:
And without aggressive inlining, with TryReadTo be inlined?  I'd assume not, in which case is there a benefit to explicitly outlinining this slow path?

Reply:
Reduced stack space zeroing from not having the 24 byte object ref `ReadOnlySequence<T>`?

Reply:
Ok, if you actually see a benefit from it.

Reply:
Split preamble - zeros 16 bytes

```asm
; Lcl frame size = 56

G_M24920_IG01:
       push     r15
       push     r14
       push     r13
       push     r12
       push     rdi
       push     rsi
       push     rbp
       push     rbx
       sub      rsp, 56
       vzeroupper 
       xor      rax, rax                         ; 16 byte zeroing
       mov      qword ptr [rsp+28H], rax
       mov      qword ptr [rsp+30H], rax
       mov      rdi, rcx
       mov      rbx, rdx
       mov      rsi, r8
       mov      ebp, r9d
```
Non-split preamble - zeros 144 bytes
```asm
; Lcl frame size = 216

G_M24920_IG01:
       push     r15
       push     r14
       push     r13
       push     r12
       push     rdi
       push     rsi
       push     rbp
       push     rbx
       sub      rsp, 216
       vzeroupper 
       vxorps   xmm4, xmm4
       mov      rax, -144                        ; 144 byte zeroing
       vmovdqa  xmmword ptr [rsp+rax+D0H], xmm4
       vmovdqa  xmmword ptr [rsp+rax+E0H], xmm4
       vmovdqa  xmmword ptr [rsp+rax+F0H], xmm4
       add      rax, 48
       jne      SHORT  -5 instr
       mov      qword ptr [rsp+D0H], rax
       mov      rbx, rcx
       mov      rbp, rdx
       mov      r14d, r9d
```

Reply:
Missed counting last `qword` one, is actually 148 bytes